From: Benjamin Drung <benjamin.drung@canonical.com>
Date: Wed, 3 Apr 2024 15:14:50 +0200
Subject: Correct timezone symlinks when using BACKWARD=backward
 PACKRATDATA=backzone

When using BACKWARD=backward PACKRATDATA=backzone the symlinks might
point to the incorrect file (e.g. Africa/Asmera points to Africa/Nairobi
instead of Africa/Asmara).

Move the link to link feature from vanguard to main dataform to produce
correct symlink (e.g. Africa/Asmera -> Africa/Asmara) in this case. Drop
those link from `backward` that would cause symlinks to symlinks.

See https://bugs.launchpad.net/ubuntu/+source/tzdata/+bug/2062522

Forwarded: https://mm.icann.org/pipermail/tz/2024-April/058853.html
---
 backward    | 8 ++++----
 ziguard.awk | 4 ++--
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/backward b/backward
index 0236751..1e5de57 100644
--- a/backward
+++ b/backward
@@ -31,7 +31,7 @@
 # Pre-1993 naming conventions
 
 # Link	TARGET			LINK-NAME	#= TARGET1
-Link	Australia/Sydney	Australia/ACT	#= Australia/Canberra
+Link	Australia/Sydney	Australia/ACT
 Link	Australia/Lord_Howe	Australia/LHI
 Link	Australia/Sydney	Australia/NSW
 Link	Australia/Darwin	Australia/North
@@ -41,7 +41,7 @@ Link	Australia/Hobart	Australia/Tasmania
 Link	Australia/Melbourne	Australia/Victoria
 Link	Australia/Perth		Australia/West
 Link	Australia/Broken_Hill	Australia/Yancowinna
-Link	America/Rio_Branco	Brazil/Acre	#= America/Porto_Acre
+Link	America/Rio_Branco	Brazil/Acre
 Link	America/Noronha		Brazil/DeNoronha
 Link	America/Sao_Paulo	Brazil/East
 Link	America/Manaus		Brazil/West
@@ -110,7 +110,7 @@ Link	America/Mazatlan	Mexico/BajaSur
 Link	America/Mexico_City	Mexico/General
 Link	Pacific/Auckland	NZ
 Link	Pacific/Chatham		NZ-CHAT
-Link	America/Denver		Navajo	#= America/Shiprock
+Link	America/Denver		Navajo
 Link	Asia/Shanghai		PRC
 Link	Europe/Warsaw		Poland
 Link	Europe/Lisbon		Portugal
@@ -309,7 +309,7 @@ Link	Africa/Nairobi		Africa/Asmera	#= Africa/Asmara
 Link	America/Nuuk		America/Godthab
 Link	Asia/Ashgabat		Asia/Ashkhabad
 Link	Asia/Kolkata		Asia/Calcutta
-Link	Asia/Shanghai		Asia/Chungking	#= Asia/Chongqing
+Link	Asia/Shanghai		Asia/Chungking
 Link	Asia/Dhaka		Asia/Dacca
 # Istanbul is in both continents.
 Link	Europe/Istanbul		Asia/Istanbul
diff --git a/ziguard.awk b/ziguard.awk
index c0acb72..d7e9ba3 100644
--- a/ziguard.awk
+++ b/ziguard.awk
@@ -336,7 +336,7 @@ function make_linkline(oldline, target, linkname, oldtarget, comment, \
   return "Link\t" target "\t" replsuffix comment
 }
 
-/^Link/ && $4 == "#=" && DATAFORM == "vanguard" {
+/^Link/ && $4 == "#=" && (DATAFORM != "rearguard") {
   $0 = make_linkline($0, $5, $3, $2)
 }
 
@@ -374,7 +374,7 @@ function cut_link_chains_short( \
 }
 
 END {
-  if (DATAFORM != "vanguard") {
+  if (DATAFORM == "rearguard") {
     cut_link_chains_short()
   }
   for (i = 1; i <= NR; i++)
